rudder-lang/src/parser.rs

```diff
@@ -190,25 +191,47 @@ macro_rules! sequence {
     ( { $($f:ident : $parser:expr;)* } => $output:expr ) => {
         move |i| {
             $(
-                let (i,$f) = $parser (i)?;
+                // save last line (cursor is updated at parser calls). context of an potential error
+                let focalized_error = get_focalized_input(i);
+                // intercept error to update its context if it should lead to a handled compilation error
+                let (i, $f) = match $parser (i) {
+                    Ok(res) => res,
+                    Err(e) => return Err(update_error_context(e, focalized_error))
+                };
```

Comment:
you could also write it: 
```
  let (j,$f) ) ... {
       ... => return Err(update_error_context(e, get_focalized_input(i)))
  };
  let i = j;
```

Then you still have access to the original i and may some day find more information to extract from it